### ISSUE TITLE
UX Bug: Why do I see some stories twice?

### ISSUE THREAD

**Issue by KristerV** [KristerV/HNSince#3]
We've been talking with @hashbackup in private about how to handle what I'll call "the new story valley of death".

Problem is with new stories that may get a lot of votes, but you'll never see, because on your visit it was 5 minutes old with 2 votes. That story will never be shown, because "you've already seen it" since it already existed when you last visited.

The current simplistic solution is to always show X hours more stories than what you'd expect. There are two problems with this approach: 1) if you visit often (multiple times a day) then you'll see the same story multiple times and 2) X in reality is not a constant variable since votes can be given for days. The current solution is basically a compromise that doesn't actually fix the problem.

We thought about it and it seems the only way to fix it is to record what stories the user has already seen in a database. This way X can be 24-h (or even 48h) and you'll always see the top stories, without them repeating.

This still feels a bit of a hack and not very elegant. But it would be much better than what we currently have.

**Comment by hashbackup** [comment]
While this is a bit confusing when it happens, HNSince saves sooooo much time for me that after using it for a while, the occasional repeated story is not a big problem.  I think for people who check HNSince very frequently, it would be a bigger problem.  I use it maybe 1-3 times per day so I'm not noticing too many repeats.

HNSince is night and day better than scrolling through HN itself a couple of times a day.

**Comment by KristerV** [comment]
I noticed another issue recently. Has only happened once, but there was a story that got lost in between visits.

As I said in another issue I have 3 devices, each with its own `last_visit`. I noticed that on mobile, where I check more often, I never saw a story that did appear on my laptop. This means that 1h is not enough. But increasing it to 2 (or more like 24h) would increase the problem of seeing the same stories by a lot.

Anyway. Current solution is still 80% there so not touching at the moment.
